public/data/data-structure-information/heap.json

```diff
@@ -0,0 +1,208 @@
+{
+  "title": "Heap",
+  "description": "",
+  "classImplementationCode": "class Heap {\n  constructor(array,typeOfHeap = 1) {\n    this.typeOfHeap = typeOfHeap;\n    this.heap = this.buildHeap(array, typeOfHeap);\n  }\n}",
+  "operations": [
+    {
+      "key": "buildHeap",
+      "methodName": "buildHeap(array, typeOfHeap)",
+      "description": "",
+      "implementationCode": " buildHeap(array, typeOfHeap) {\n   \n      const firstParentIdx = Math.floor((array.length - 2) / 2);\n      \n      for(let currentIdx = firstParentIdx; currentIdx >=0; currentIdx--) {\n         this.shiftDown(currentIdx, array.length - 1, array, typeOfHeap);\n      }\n  \n      return array; \n  }",
+      "parameters": [
+        {
+          "name": "array",
+          "description": "Array of numbers for which heap you want to create heap"
+        },
+        {
+          "name": "typeOfHeap",
```

Comment:
Just incase if possible: instead of type can we use comparator function like greater/min/max/something such that we can use. that will also helping in the building and swapping the items.

Reply:
@surya76657 Yes that will be possible but since this was a flag an if you see I have used this flag for the same purpose for swapping and other conditions

Reply:
Also we have add a custom comparator like xor or sum multiply or sum custom logic over there like near to origin and all like that

Reply:
Will enhance it further

Reply:
awesome. thanks man!!!